File: FleetMaintenance.Domain/Entities/MaintenanceType.cs
namespace FleetMaintenance.Domain.Entities
{
    public class MaintenanceType
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public ICollection<MaintenanceRecord> MaintenanceRecords { get; set; }
            = new List<MaintenanceRecord>();
    }
}

File: fleet-maintenance-client/src/types/maintenanceType.ts
export interface MaintenanceType {
  id: number;
  name: string;
  description: string | null;
}

export interface CreateMaintenanceTypeRequest {
  name: string;
  description?: string | null;
}

export interface UpdateMaintenanceTypeRequest {
  name?: string;
  description?: string | null;
}